packages/playground/src/components/webhooks/create-webhooks-form-fields.tsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react'
+import { Input, Textarea, Checkbox, StackedList, Switch, RadioGroup, RadioGroupItem } from '@harnessio/canary'
+import { FormFieldSet, MessageTheme } from '../../index'
+import { WebhookEvent, FieldProps, BranchEvents, TagEvents, PREvents, EventTypes } from './types'
+
+export const WebhookToggleField: React.FC<FieldProps> = ({ register, watch, setValue }) => (
+  <StackedList.Root className="border-none">
+    <StackedList.Item disableHover isHeader>
+      <StackedList.Field
+        title="Enable the webhook"
+        description="We will deliver event details when this hook is triggered"
+      />
+      <StackedList.Field
+        label
+        secondary
+        title={
+          <div className="flex gap-1.5 items-center justify-end cursor-pointer">
+            <Switch
+              {...register!('state')}
+              checked={watch!('state')}
+              onCheckedChange={() => setValue!('state', !watch!('state'))}
+            />
+          </div>
+        }
+        right
+      />
+    </StackedList.Item>
+  </StackedList.Root>
+)
+
+export const WebhookNameField: React.FC<FieldProps & { disabled: boolean }> = ({ register, errors, disabled }) => (
+  <FormFieldSet.ControlGroup>
+    <FormFieldSet.Label htmlFor="name" required>
+      Name
+    </FormFieldSet.Label>
+    <Input id="name" {...register!('name')} placeholder="Name your webhook" autoFocus disabled={disabled} />
+    {errors!.name && (
+      <FormFieldSet.Message theme={MessageTheme.ERROR}>{errors!.name.message?.toString()}</FormFieldSet.Message>
+    )}
+  </FormFieldSet.ControlGroup>
+)
+
+export const WebhookDescriptionField: React.FC<FieldProps> = ({ register, errors }) => (
+  <FormFieldSet.ControlGroup>
+    <FormFieldSet.Label htmlFor="description" required>
+      Description
+    </FormFieldSet.Label>
+    <Textarea id="description" {...register!('description')} placeholder="Enter a description of this rule..." />
+    {errors!.description && (
+      <FormFieldSet.Message theme={MessageTheme.ERROR}>{errors!.description.message?.toString()}</FormFieldSet.Message>
+    )}
+  </FormFieldSet.ControlGroup>
+)
+
+export const WebhookPayloadUrlField: React.FC<FieldProps> = ({ register, errors }) => (
+  <FormFieldSet.ControlGroup>
+    <FormFieldSet.Label htmlFor="payloadUrl" required>
+      Payload URL
+    </FormFieldSet.Label>
+    <Input id="payloadUrl" {...register!('payloadUrl')} placeholder="https://example.com/harness" />
+    {errors!.payloadUrl && (
+      <FormFieldSet.Message theme={MessageTheme.ERROR}>{errors!.payloadUrl.message?.toString()}</FormFieldSet.Message>
+    )}
+  </FormFieldSet.ControlGroup>
+)
+
+export const WebhookSecretField: React.FC<FieldProps> = ({ register, errors }) => (
+  <FormFieldSet.ControlGroup>
+    <FormFieldSet.Label htmlFor="secret" required>
```

Comment:
should secret be required? github allows webhooks without a secret. does gitness not allow the same?